### ISSUE TITLE
Plot planetary system configuration

### ISSUE THREAD

**Issue by DanielAndreasen** [DanielAndreasen/SWEETer-Cat#60]
With inspiration from [this](http://mpld3.github.io/examples/custom_plugin.html) example from `mpld3`, I was thinking to have a lower plot with maybe `teff` vs. `semi-major-axis`. When we hover over a point, the upper plot will show on the x-axis the distance to all the exoplanets in the system and the location of the habitable zone. In this example the y-axis is irrelevant, but could maybe somehow symbolize the size of the objects (star and planet(s)).

**Comment by jason-neal** [comment]
Yea that is pretty neat graph and would make for an interesting view. 
How do you envision these interacting with the other plots? Would all these different plot format/styles ideas be on separate pages? Or do you envision the different sub plots be "activated" using 'checkboxes'. 
I imagine that many different interacting plots could get overwhelming all at once. (eg. 4 panels mpld3 + distribution historgrams + planet configuration)

**Comment by DanielAndreasen** [comment]
So I do admit I might have been carried a little away with this. It's just so cool with interactive plots.
However, I had some time to think about it. How about making that "system plot" in the detailed view of the stars? Maybe span it on the top of the page. Give the star a colour based on the temperature, the sizes should be representative of the radii, and with the HZ as well.

I think you are right that we should not suddenly have 4 pages for plotting stuff. That is beyond the scope of this project (in my opinion).

**Comment by DanielAndreasen** [comment]
I added a plot to the top of the star-detail page (not visible for all stars, but most I think) with commit  f16e698.
The size of the star is updated for each star using its radius as a variable.
Things to do:

- [ ] The same thing should be done with the planets.
- [ ] The color of star/planet(s) should also represent the temperature.
- [ ] Information during hoverering with mouse

**Comment by jason-neal** [comment]
[Here](http://bokeh.pydata.org/en/latest/docs/user_guide/styling.html#responsive-dimensions) is a link for responsive bokeh plots. I haven't yet found a similar link for mpld3

**Comment by jason-neal** [comment]
I have been looking at the star detail plot a bit and have a number of notes/ideas. ( I know it is not finished yet).

Currently, the radius scaling of the planet is not done/does not make sense.
One check is `HD 103720 ` and  `HD 189733`, they have similar temperature stars. Currently, the larger and closer planet has the smaller planet dot of the two. I assume this should be the other way around.

Currently, a planet is fixed at the same location (unless multiple). That doesn't easily help understand its distance. 
One idea would be to have a set of fixed window sizes that the planet can fall in. e.g. 0.05 AU, 0.25 AU, 1 AU etc. so that the planet moves around a bit more. This would need to make the distance units/scale more visible, maybe even mark edge the smaller bins. 
 e.g  `(star)....| 0.05.............. | 0.25.............(planet)...........| 0.75 ` 

I don't know how useful it would be to include the distances/sizes of the solar system planets (if they fall in the range spanned by the current star), like "Oh btw here is Lil' Mercury"

Do you want a label on the distance axes of ("AU")?

Would an indicator on the plot that there is a planet not placed/shown?  e.g. there are three planets but we only have the small for one of them. It could have 2 extra planets along the bottom or something, either shapes for them or a simple counter.  `GJ 849` is one example of this.

The Hz is a nice touch.

All of these are not nesscessary but came to mind when looking at the stars.

**Comment by DanielAndreasen** [comment]
Let me try to reply to your ideas/questions in the same order you asked them :)

> Currently, the radius scaling of the planet is not done/does not make sense.
One check is HD 103720 and HD 189733, they have similar temperature stars. Currently, the larger and closer planet has the smaller planet dot of the two. I assume this should be the other way around.

I think I understand what you are writing. Mass of `HD 189733 b` is higher than `HD 103720 b` yet the latter is bigger. If you look into the code, I simple assume a density for a planet in order to calculate the radius in the cases where the radius is not known. Better ideas are welcome. But maybe we can include the assumed density somewhere.

> * Currently, a planet is fixed at the same location (unless multiple). That doesn't easily help understand its distance.
One idea would be to have a set of fixed window sizes that the planet can fall in. e.g. 0.05 AU, 0.25 AU, 1 AU etc. so that the planet moves around a bit more. This would need to make the distance units/scale more visible, maybe even mark edge the smaller bins.
e.g `(star)....| 0.05.............. | 0.25.............(planet)...........| 0.75`
> * I don't know how useful it would be to include the distances/sizes of the solar system planets (if they fall in the range spanned by the current star), like "Oh btw here is Lil' Mercury"
> * Do you want a label on the distance axes of ("AU")?

I like these ideas a lot!

> Would an indicator on the plot that there is a planet not placed/shown? e.g. there are three planets but we only have the `sma` for one of them. It could have 2 extra planets along the bottom or something, either shapes for them or a simple counter. GJ 849 is one example of this.

I think we somehow need to tell that some planets are missing in the plot. I think a text in the upper right corner might be a way to go about it

> The Hz is a nice touch.

Thanks :) The colours of the planets represents the distance to the HZ. Between star and HZ they are red-green-yellow, and after they should be more green to blue. See the colormaps [here](https://matplotlib.org/users/colormaps.html).

A last thing I was thinking about it to make the background of the plot black or at least dark. Probably it is something we play around with and see how well it works.

**Comment by jason-neal** [comment]
Yes I think including the assumed density somewhere would be a good idea. Maybe it can go in the corner with the missing planets, or as a footnote?  It may get tricky if there is more than one planet with differently assumed radii.

With the background, I wonder if having the background the same color as the menu bar be dark enough?

**Comment by jason-neal** [comment]
I also found some stars that were huge! Taking up about 1/4 of the plot. I don't have a number on hand as a example. I think they were the two strangely named stars on the current homepage starting TYC or BD.

So there maybe an issue with the star scaling also.

**Comment by DanielAndreasen** [comment]
As we talked about. Include size of Earth, Neptune, Jupiter, and the Sun for reference where it makes sense.

**Comment by DanielAndreasen** [comment]
There is a name showing now in the plot if a planet is missing with commit 1b082b865e11f. It could use some more styling, but it works. An example is [GJ 849](http://localhost:5000/star/GJ%20849/)

**Comment by DanielAndreasen** [comment]
I think this issue is no longer relevant (I didn't read it all) since it the current detailed pages are in a good stage.
